### PR TITLE
with-apollo-auth updated for Apollo 2.0

### PR DESCRIPTION
**Pull request by DevSpeak**
Took a stab at trying to update it for Apollo 2.0

It's my first time trying to contribute, hope it's useful.

### CODE REVIEW

**Comment by timneutkens** [comment]
@ads1018 can you check this

**Review by adamsoffer — APPROVED** [review]
Just tested on my machine. Everything looks great. Great work. Thanks so much!

**Comment by DevSpeak** [comment]
Thanks @ads1018 .
I'm not sure if I did something wrong when I tried to do "Resolve conflicts".

Please notify me if I have to change something :)

File: examples/with-apollo-auth/lib/withData.js [hunk 1]
```diff
@@ -46,28 +47,34 @@ export default ComposedComponent => {
...
+        Head.rewind()
 
         // Extract query data from the Apollo's store
-        const state = apollo.getInitialState()
-
         serverState = {
-          apollo: { // Make sure to only include Apollo's data state
-            data: state.data
+          apollo: {
+            data: apollo.cache.extract()
```
Comment (danistefanovic) [inline examples/with-apollo-auth/lib/withData.js]: Can be simplified to just:

```js
serverState = apollo.cache.extract();
```
The change in the constructor can then be reverted:

```diff
-     this.apollo = initApollo(this.props.serverState.data, {
+     this.apollo = initApollo(this.props.serverState, {
```

**Comment by DevSpeak** [comment]
@danistefanovic Thanks for that, looks way better.
Updated it.
_The commit history looks idiotic because I'm learning to use pull requests._

**Review by timneutkens — APPROVED** [review]
Thanks!

**Pull request merged**